ntb_hw_switchtec.c

```diff
@@ -1557,6 +1555,8 @@ static void switchtec_ntb_deinit_crosslink(struct switchtec_ntb *sndev)
 		pci_iounmap(sndev->stdev->pdev, sndev->mmio_xlink_ntinfo_win);
 
 	switchtec_ntb_cleanup_crosslink(sndev);
+
+	sndev->nr_rsvd_luts -= 3;
```

Comment:
Is this really necessary? If deinit is being called the sndev structure is going to be deallocated soon and this value will not have any importance. It can't be reinitialized ever after this can it?

Actually, I'm really not sure why this patch exists, the commit message doesn't explain that.

Reply:
Not necessary function wise, just to make the init and deinit symmetric, it cannot be reinitialized. Will remove it. 

As two of the three reserved LUTs are initialized after probe, to prevent these two LUT slots from unintentionally allocated by user code, I reserve all three LUTs by increasing _nr_rsvd_luts_ by three in the init code rather than increasing them when they get initialized during link up handshake. Will add more description in the commit message.